tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -87,8 +92,26 @@ class AppFilterImpl(
     } else {
       apps
     }
+
+   // If `any` is specified as logicFilter config, then store the value in variable which is used
+   // later for final computation.
+   val appNameFilteredForDisjunction = if (appArgs.applicationName.isSupplied &&
+       filterAppName.nonEmpty && appArgs.any()
+       && logicFilter.contains("any")) {
+      appNameFiltered
+    } else {
+       Seq.empty[AppFilterReturnParameters]
+    }
+
     val userNameFiltered = if (appArgs.userName.isSupplied && userName.nonEmpty) {
-      val filtered = appNameFiltered.filter { app =>
+      val appNamelogicFiltered = if (appArgs.any() &&
+          logicFilter.contains("any")) {
+        apps
+      } else {
+        appNameFiltered
+      }
+
+      val filtered = appNamelogicFiltered.filter { app =>
```

Comment:
This could be simplified by replacing 

```scala
val filtered = appNamelogicFiltered.filter { app =>
  if (app.appInfo.isDefined) {
    app.appInfo.get.userName.contains(userName)
  } else {
    false
  }
}
filtered
```

with 

```scala
appNamelogicFiltered.filter(_.appInfo.exists(_.userName.contains(userName)))
```